### PR TITLE
More `Cms` namespace type hinting/clean up

### PR DESCRIPTION
**Pull request by distantnative**

### CODE REVIEW

File: src/Cms/LanguageRouter.php [hunk 1]
```diff
@@ -106,26 +91,32 @@ public function routes(): array
 	 * Wrapper around the Router::call method
 	 * that injects the Language instance and
 	 * if needed also the Page as arguments.
-	 *
-	 * @param string|null $path
-	 * @return mixed
 	 */
-	public function call(string $path = null)
+	public function call(string $path = null): mixed
```
Comment (afbora) [inline src/Cms/LanguageRouter.php]:
```suggestion
	public function call(string|null $path = null): mixed
```

File: src/Cms/LanguageRouter.php [hunk 2]
```diff
@@ -106,26 +91,32 @@ public function routes(): array
...
 	{
-		$language = $this->language;
-		$kirby    = $language->kirby();
-		$router   = new Router($this->routes());
+		$language       = $this->language;
+		$kirby          = $language->kirby();
+		$this->router ??= new Router($this->routes());
```
Comment (afbora) [inline src/Cms/LanguageRouter.php:99]: Caching router instance is intended?

File: src/Cms/LanguageRoutes.php [hunk 3]
```diff
@@ -73,23 +67,28 @@ public static function fallback(App $kirby): array
...
+
 						if (
 							$kirby->option('languages.detect') === true &&
-							$page->translation($kirby->detectedLanguage()->code())->exists() === true
+							$page->translation($lang)->exists() === true
 						) {
 							return $kirby
 								->response()
-								->redirect($page->url($kirby->detectedLanguage()->code()));
+								->redirect($lang);
```
Comment (afbora) [inline src/Cms/LanguageRoutes.php]:
```suggestion
								->redirect($page->url($lang));
```

File: src/Cms/Languages.php [hunk 4]
```diff
@@ -24,12 +24,13 @@ class Languages extends Collection
...
 	 *
-	 * @param array $objects `Kirby\Cms\Language` objects
 	 * @param null $parent
 	 * @throws \Kirby\Exception\DuplicateException
 	 */
-	public function __construct($objects = [], $parent = null)
-	{
+	public function __construct(
+		array $objects = [],
+		$parent = null
```
Comment (afbora) [inline src/Cms/Languages.php:32]:
```suggestion
		object|null $parent = null
```

**Review by afbora — COMMENTED** [review]
I can only find above minor things from mobile 😅

File: src/Cms/LanguageRouter.php (hunk 2 above)
Comment (distantnative) [reply src/Cms/LanguageRouter.php:99]: We always had the property in the class but it was never used, thought this probably was supposed to always be cached

File: src/Cms/Languages.php (hunk 4 above)
Comment (distantnative) [reply src/Cms/Languages.php:32]: According to the doc block, we actually don't want this parameter to be set to anything

**Review by afbora — APPROVED** [review]

**Pull request merged**